proxy/logging/LogFilter.h

```diff
@@ -382,6 +382,61 @@ LogFilterString::_checkCondition(OperatorFunction f, const char *field_value, si
   return retVal;
 }
 
+static const char *
+findPatternFromParamName(const char *lookup_query_param, const char *pattern)
+{
+  const char *p1 = strstr(lookup_query_param, pattern);
+  while (p1) {
+    // wipe pattern in param name, need search again if find pattern in param value
+    const char *p10 = strchr(p1, '=');
+    if (!p10) {
+      // no "=" after p1, means p1 is not in the param name, and no more param after it
+      p1 = nullptr;
+      break;
+    }
+    const char *p11 = strchr(p1, '&');
+    if (p11 && p10 > p11) {
+      //"=" is after "&" followd by p1, means p1 is not in the param name
+      p1 = strstr(p11, pattern);
+      continue;
+    }
+    // ensure p1 is in the param name now
+    break;
+  }
+  return p1;
+}
+
+static void
+updatePatternForFieldValue(char **field, const char *p1, int field_pos, char *buf_dest)
+{
+  char tmp_text[strlen(buf_dest) + 10];
+  char *temp_text = tmp_text;
+  memcpy(temp_text, buf_dest, (p1 - buf_dest));
+  temp_text += (p1 - buf_dest);
+  const char *p2 = strchr(p1, '=');
+  if (p2) {
+    p2++;
+    memcpy(temp_text, p1, (p2 - p1));
+    temp_text += (p2 - p1);
+    const char *p3 = strchr(p2, '&');
+    if (p3) {
+      for (int i = 0; i < (p3 - p2); i++)
+        temp_text[i] = 'X';
+      temp_text += (p3 - p2);
+      memcpy(temp_text, p3, ((buf_dest + strlen(buf_dest)) - p3));
+    } else {
+      for (int i = 0; i < ((buf_dest + strlen(buf_dest)) - p2); i++) {
```

Comment:
a performance killer to have strlen() in the for loop condition.

Reply:
Good catch. I believe most compilers these days are smart enough to optimise this sort of usage already, so in practice it is probably not being evaluated each iteration anyway.

Regardless, +1 to fix it.

FWIW, this was an existing piece of code lying around forever (I blame myself :( ) and not something that @whutwhu specifically changed in this PR (other than just modularising it).

Reply:
I didn't notice it from original code, and thank you @zizhong for your careful review!